design/README.md

```diff
@@ -1,48 +1,75 @@
 # System Architecture
```

Comment:
This is called "System Architecture" on its own page but "System Design" in the table of contents. Let's match titles!